src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -258,8 +258,8 @@ export function VislibVisualizationsPieChartProvider(Private) {
               const current = d.position;
               if (point) {
                 const horizontalConflict = (point.left < 0 && current.left < 0) || (point.left > 0 && current.left > 0);
-                const verticalConflict = (point.top > current.top && point.top <= current.bottom) ||
-                                          (point.top < current.top && point.bottom >= current.top);
+                const verticalConflict = (point.top >= current.top && point.top <= current.bottom) ||
```

Comment:
This fix does solve the conflict but It hide the label which conflict with other which looks like a bug. (missing label) 
What it should do is to try to find a better place to be shown. Usually there is still plenty of space

Reply:
The same happens with levels on the same level as well, if conflict is detected its just hidden. Could you open a separate enhancement request for trying to find new position for the label if it is in conflict with another ? (i had this implemented but we decided to move it out of the original PR as the algorithm i was using was resource intensive)

Reply:
where can I see the code for trying to find new position for the label if it is in conflict with another?

Reply:
In addition ,I opened feature request: https://github.com/elastic/kibana/issues/15571

Reply:
thanks @ofirmgr 
this is the PR: https://github.com/elastic/kibana/pull/12174 however it seems i forced pushed after rebasing on master and that commit is no longer available.

basically it tried to move the label up/down if conflict was detected and ran the conflict resolution again.